libraries/profile.rb

```diff
@@ -46,20 +47,35 @@ class ComplianceProfile < Chef::Resource
     converge_by 'fetch compliance profile' do
       o, p = normalize_owner_profile
       Chef::Log.info "Fetch compliance profile #{o}/#{p}"
-      url = construct_url("organizations/#{org}/owners/#{o}/compliance/#{p}/tar")
-
-      Chef::Config[:verify_api_cert] = false
-      Chef::Config[:ssl_verify_mode] = :verify_none
-
-      rest = Chef::ServerAPI.new(url, Chef::Config)
-      tf = rest.binmode_streaming_request(url)
-
-      # don't delete temp file on GC
-      ObjectSpace.undefine_finalizer(tf)
 
       path = tar_path
       directory(::Pathname.new(path).dirname.to_s).run_action(:create)
 
+      if token # go direct
+        reqpath ="owners/#{o}/compliance/#{p}/tar"
+        url = construct_url(reqpath, server)
+        puts "URL: #{url}"
+
+        tf = Tempfile.new('foo', Dir.tmpdir, 'wb+')
+        tf.binmode
+        Net::HTTP.start(url.host, url.port) do |http|
+          http.use_ssl = url.scheme == 'https'
+          http.verify_mode = OpenSSL::SSL::VERIFY_NONE # FIXME
```

Comment:
I'm afraid I don't get it. It's not boolean but a string property that is `nil` if unset and some string if it was provided.
So currently, it's a property of a resource. You can use attributes to set it in your _recipe_ ([see here](https://github.com/chef-cookbooks/audit/pull/20/files#diff-ee5748cee3a2d43791dd1e832193194aR29)) -- I haven't had a need to touch the default recipe here.

Having a resource break its interface and _reach out_ into node attributes doesn't seem right to me...